lib/oauthservertoserver.js

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2023 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+"use strict";
+
+const fetch = require("@adobe/node-fetch-retry");
+const FormData = require('form-data');
+
+/**
+ * Check if an integration configuration is an OAuth Server to Server integration
+ * 
+ * @param {*} integration Integration configuration
+ * @returns {boolean} True if the integration is an OAuth Server to Server integration
+ */
+function isOAuthServerToServerIntegration(integration) {
+    return integration ? integration.TYPE === 'oauthservertoserver' : false;
+}
+
+/**
+ * Validate an OAuth Server to Server integration configuration
+ * 
+ * @param {*} integration Integration configuration
+ * @returns {boolean} True if the integration is a valid OAuth Server to Server integration
+ */
+function validateOAuthServerToServerIntegration(integration) {
```

Comment:
I see that you are checking for the presence of the following fields. How does this confirm that its oauth? What's the diff between oauth and jwt config?

- ORGID
- CLIENT_SECRETS
- CLIENT_ID
- SCOPES
- TECHNICAL_ACCOUNT_ID
- TECHNICAL_ACCOUNT_EMAIL

Reply:
there are sample oauth/jwt files in `resources` folder